Application/EdFi.Ods.Api/Security/Authorization/AuthorizationSegmentSqlProviderBase.cs

```diff
@@ -80,12 +79,54 @@ public AuthorizationSegmentSqlProviderBase(IAuthorizationViewsProvider authoriza
                     }
 
                     // Perform defensive checks against the remote possibility of SQL injection attack
-                    ValidateTableNameParts(claimEndpointName, subjectEndpointName, authorizationSegment.AuthorizationPathModifier);
+                    ValidateTableNameParts(
+                        claimEndpointName, subjectEndpointName, authorizationSegment.AuthorizationPathModifier);
 
-                    string derivedAuthorizationViewName = ViewNameHelper.GetFullyQualifiedAuthorizationViewName(
-                        subjectEndpointName,
-                        claimEndpointName,
-                        authorizationSegment.AuthorizationPathModifier);
+                    var entities = new List<string>()
+                    {
+                        "ThroughEdOrgAssociation",
+                        "ParentUSI",
+                        "StaffUSI"
+                    };
+
+                    string derivedAuthorizationViewName;
+
+                    // When subject Endpoint or Claim Endpoint ends with StudentUSI and also not ends with ThroughEdOrgAssociation
+                    // use new Student view 
+                    if ((subjectEndpointName.EndsWith("StudentUSI") || claimEndpointName.EndsWith("StudentUSI")) &&
```

Comment:
I moved the same logic in ClaimsAuthorizationSegment class since that's where ClaimsEndpoints are getting added from ClaimsMustBeAssociatedWith(params SegmentProperty[] segmentProperties) method in AuthorizationBuilder class .